File: Doubao_MarsCode/java/MC0039/Main.java
package MC0039;

/**
 * 39. 计算从位置x到y的最少步数
 */
public class Main {
    public static int solution(int x, int y) {
        int dist = Math.abs(x - y);

        int flag = 0;
        int length = 1;
        int steps = 0;
        while (dist > 0) {
            if (flag == 2) {
                length++;
                flag = 0;
            }
            dist -= length;
            flag++;
            steps++;
        }

        return steps;
    }

    public static void main(String[] args) {
        // You can add more test cases here
        System.out.println(solution(12, 6) == 4);
        System.out.println(solution(34, 45) == 6);
        System.out.println(solution(50, 30) == 8);
    }
}

File: Doubao_MarsCode/go/MC0039/main.go
package main

import (
	"fmt"
	"math"
)

func solution(x int, y int) int {
	dist := int(math.Abs(float64(x - y)))

	flag := 0
	length := 1
	steps := 0
	for dist > 0 {
		if flag == 2 {
			length++
			flag = 0
		}
		dist -= length
		flag++
		steps++
	}

	return steps
}

func main() {
	//  You can add more test cases here
	fmt.Println(solution(12, 6) == 4)
	fmt.Println(solution(34, 45) == 6)
	fmt.Println(solution(50, 30) == 8)
}
